TodoApp.ConsoleApp/Repositories/XMLRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace TodoApp.ConsoleApp.Repositories
+{
+    abstract class XMLRepository<T> : IRepository<T>
+    {
+        const string path = "../../tasks.xml";
+        public virtual XElement ParentElement { get; protected set; }
+
+        protected XName ElementName { get; private set; }
+
+        protected abstract Func<XElement, T> Selector { get; }
+
+        protected abstract void SetXElementValue(T model, XElement element);
+
+        protected abstract XElement CreateXElement(T model);
+
+        protected abstract object GetEntityId(T entidade);
```

Comment:
_Just a note: The idea of `protected abstract` properties & methods is that they are specific for each sub-class.
In this case you have to use such methods & properties if they contain some specific functionality for the `TodoRepository`._